src/finiteVolume/cellCentred/auxiliary/coNum.cpp

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+
+#include <limits>
+
+#include "NeoFOAM/core/info.hpp"
+#include "NeoFOAM/core/parallelAlgorithms.hpp"
+#include "NeoFOAM/finiteVolume/cellCentred/auxiliary/coNum.hpp"
+
+namespace NeoFOAM::finiteVolume::cellCentred
+{
+
+scalar computeCoNum(const SurfaceField<scalar>& faceFlux, const scalar dt)
+{
+    const UnstructuredMesh& mesh = faceFlux.mesh();
+    const auto exec = faceFlux.exec();
+    VolumeField<scalar> phi(exec, "phi", mesh, createCalculatedBCs<VolumeBoundary<scalar>>(mesh));
+    fill(phi.internalField(), 0.0);
+    const auto surfFaceCells = mesh.boundaryMesh().faceCells().span();
+
+    const auto volPhi = phi.internalField().span();
+    const auto surfOwner = mesh.faceOwner().span();
+    const auto surfNeighbour = mesh.faceNeighbour().span();
+    const auto surfFaceFlux = faceFlux.internalField().span();
```

Comment:
you could use spans() and structured bindings here (untested)
```suggestion
    const auto [volPhi,surfOwner, surfNeighbour,surfFaceFlux] = spans(phi.internalField(), mesh.faceOwner(), mesh.faceNeighbour(), faceFlux.internalField())
```